File: src/main/java/net/stefanfuchs/jslt/intellij/language/JsltFileType.java
package net.stefanfuchs.jslt.intellij.language;


import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class JsltFileType extends LanguageFileType {

  public static final JsltFileType INSTANCE = new JsltFileType();

  private JsltFileType() {
    super(JsltLanguage.INSTANCE);
  }

  @NotNull
  @Override
  public String getName() {
    return "Simple File";
  }

  @NotNull
  @Override
  public String getDescription() {
    return "Simple language file";
  }

  @NotNull
  @Override
  public String getDefaultExtension() {
    return "simple";
  }

  @Nullable
  @Override
  public Icon getIcon() {
    return JsltIcons.FileType;
  }

}

File: src/main/kotlin/net/stefanfuchs/jslt/intellij/language/JsltFileType.kt
package net.stefanfuchs.jslt.intellij.language

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

class JsltFileType private constructor() : LanguageFileType(JsltLanguage) {
    override fun getName(): String {
        return "Jslt File"
    }

    override fun getDescription(): String {
        return "JSLT transformation file"
    }

    override fun getDefaultExtension(): String {
        return "jslt"
    }

    override fun getIcon(): Icon {
        return JsltIcons.FileType
    }

    companion object {
        val INSTANCE = JsltFileType()
    }
}